crate/server/src/kms_server.rs

```diff
@@ -185,33 +185,35 @@ pub async fn prepare_kms_server(
     let google_cse_jwks = enable_google_cse.then_some(google_cse::list_jwks_uri());
 
     // Determine if JWT Auth should be used for authentication.
-    let (use_jwt_auth, jwt_config, jwks_manager) = if let Some(jwt_issuer_uri) =
-        &kms_server.params.jwt_issuer_uri
-    {
-        // Prepare all the needed URIs
-        let jwks_uri = JwtAuthConfig::uri(jwt_issuer_uri, kms_server.params.jwks_uri.as_deref());
-        let uris = google_cse_jwks.map_or_else(
-            || vec![jwks_uri.clone()],
-            |mut google_uris| {
-                google_uris.push(jwks_uri.clone());
-                google_uris
-            },
-        );
+    let (use_jwt_auth, jwt_config, jwks_manager) =
+        if let Some(jwt_config) = &kms_server.params.jwt_config {
+            // Prepare all the needed URIs from all the configured Identity Providers
+            let mut all_jwks_uris: Vec<_> = jwt_config
+                .iter()
+                .map(|idp_config| {
+                    JwtAuthConfig::uri(&idp_config.jwt_issuer_uri, Some(&idp_config.jwks_uri))
```

Comment:
Isn't this option the one you removed from the parsing? Maybe you did need to keep it after all?

Reply:
Yes - now if jwks is not provided, it's filled hre, using this method.